factcast-schema-registry-cli/src/main/resources/site/hugo-theme-docdock-master/static/js/learn.js

```diff
@@ -0,0 +1,337 @@
+
+    $('#toc-menu').hover(function() {
+        $('.progress').stop(true, false, true).fadeToggle(100);
+    });
+
+    $('.progress').hover(function() {
+        $('.progress').stop(true, false, true).fadeToggle(100);
+    });
+});
+
+jQuery(window).on('load', function() {
+
+    function adjustForScrollbar() {
+        if ((parseInt(jQuery('#body-inner').height()) + 83) >= jQuery('#body').height()) {
+            jQuery('.nav.nav-next').css({ 'margin-right': getScrollBarWidth() });
+        } else {
+            jQuery('.nav.nav-next').css({ 'margin-right': 0 });
+        }
+    }
+
+    // adjust sidebar for scrollbar
+    adjustForScrollbar();
+
+    jQuery(window).smartresize(function() {
+        adjustForScrollbar();
+    });
+
+    // store this page in session
+    sessionStorage.setItem(jQuery('body').data('url'), 1);
+
+    // loop through the sessionStorage and see if something should be marked as visited
+    for (var url in sessionStorage) {
+        if (sessionStorage.getItem(url) == 1) jQuery('[data-nav-id="' + url + '"]').addClass('visited');
+    }
+
+
+    $(".highlightable").highlight(sessionStorage.getItem('search-value'), { element: 'mark' });
+});
+
+$(function() {
+    $('a[rel="lightbox"]').featherlight({
+        root: 'section#body'
+    });
+});
+
+jQuery.extend({
+    highlight: function(node, re, nodeName, className) {
+        if (node.nodeType === 3) {
+            var match = node.data.match(re);
+            if (match) {
+                var highlight = document.createElement(nodeName || 'span');
+                highlight.className = className || 'highlight';
+                var wordNode = node.splitText(match.index);
+                wordNode.splitText(match[0].length);
+                var wordClone = wordNode.cloneNode(true);
+                highlight.appendChild(wordClone);
+                wordNode.parentNode.replaceChild(highlight, wordNode);
+                return 1; //skip added node in parent
+            }
+        } else if ((node.nodeType === 1 && node.childNodes) && // only element nodes that have children
+            !/(script|style)/i.test(node.tagName) && // ignore script and style nodes
+            !(node.tagName === nodeName.toUpperCase() && node.className === className)) { // skip if already highlighted
+            for (var i = 0; i < node.childNodes.length; i++) {
+                i += jQuery.highlight(node.childNodes[i], re, nodeName, className);
+            }
+        }
+        return 0;
+    }
+});
+
+jQuery.fn.unhighlight = function(options) {
+    var settings = {
+        className: 'highlight',
+        element: 'span'
+    };
+    jQuery.extend(settings, options);
+
+    return this.find(settings.element + "." + settings.className).each(function() {
+        var parent = this.parentNode;
+        parent.replaceChild(this.firstChild, this);
+        parent.normalize();
+    }).end();
+};
+
+jQuery.fn.highlight = function(words, options) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A function should not mix return statements with and without a result.](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=4955738)